services/src/main/java/org/keycloak/broker/oidc/mappers/AbstractJsonUserAttributeMapper.java

```diff
@@ -136,7 +137,9 @@ public void preprocessFederatedIdentity(KeycloakSession session, RealmModel real
 
 		Object value = getJsonValue(mapperModel, context);
 		if (value != null) {
-			if (value instanceof List) {
+			if (EMAIL_VERIFIED.equalsIgnoreCase(attribute)) {
```

Comment:
I am not sure why there is a need for special handling of "emailVerified" attribute on this particular place? As I can see that none of other "special" attributes" (username, email, firstName, lastname) is handled here in this class.

IMO it will be better to make sure that all the places in the model classes like `org.keycloak.models.jpa.UserAdapter` and `org.keycloak.models.map.user.MapUserAdapter` are updated to make sure that:
`user.setAttribute("emailVerified", "true")` works as expected and calls `setEmailVerified` internally inside the UserAdapter itself. Similarly will be good if all other methods like `user.setSingleAttribute`, `user.getAttribute`, `user.getFirstAttribute` are updated to properly handle "emailVerified" attribute and delegate to the particular "setEmailVerified" or "getEmailVerified" property internally. I see you updated just single place in these classes, but not all places where other special attributes are handled.

In other words, will be good to handle emailVerified in the same places in the codebase when all other mess related to other "special attributes" (username, email, firstName, lastname) is handled. But will be good to avoid introduce more mess to other places :-)

I guess that if you properly update model classes, there won't be even need to update any brokering stuff at all (not even introduce `BrokeredIdentityContext.setEmailVerified()` and `BrokeredIdentityContext.getEmailVerified()` methods. As calling of `user.setAttribute("emailVerified", "true")` will be effectively same as `user.setEmailVerified(true)` and similarly for getters. But maybe I am too optimistic here :-)

Reply:
This if is needed here. See my general comment.